DOC mention setting `SCIPY_ARRAY_API=1` in Array API support document (#30513)
scikit-learn/scikit-learn: doc/modules/array_api.rst

The `Array API <https://data-apis.org/array-api/latest/>`_ specification defines
a standard API for all array manipulation libraries with a NumPy-like API.
Scikit-learn's Array API support requires
`array-api-compat <https://github.com/data-apis/array-api-compat>`__ to be installed,

> and the environment variable `SCIPY_ARRAY_API` must be set to `1`:

Reviewer: Note this is in principle a temporary thing in scipy while array API support is being worked on but I think this is worth mentioning anyway because this temporary thing my last a bit of time.

> and the environment variable `SCIPY_ARRAY_API` must be set to `1`:

Reviewer: Maybe mention this as well:
```suggestion
and the environment variable `SCIPY_ARRAY_API` must be set to `1`:
before importing scipy and scikit-learn.
```

.. prompt:: bash $

>    SCIPY_ARRAY_API=1

Reviewer: Use `export` as in the [scipy doc](https://docs.scipy.org/doc/scipy/dev/api-dev/array_api.html#using-array-api-standard-support):
```suggestion
   export SCIPY_ARRAY_API=1
```
One way it can matter because `export` also affect subprocess
```
SCIPY_ARRAY_API=1
# without export SCIPY_ARRAY_API is not seen as set in python script (i.e. subprocess of the current shell process)
python script_using_array_api.py
```

The `Array API <https://data-apis.org/array-api/latest/>`_ specification defines
a standard API for all array manipulation libraries with a NumPy-like API.
Scikit-learn's Array API support requires
`array-api-compat <https://github.com/data-apis/array-api-compat>`__ to be installed,
and the environment variable `SCIPY_ARRAY_API` must be set to `1` before importing
`scipy` and `scikit-learn`:

.. prompt:: bash $

   export SCIPY_ARRAY_API=1

Please note that this environment variable is intended for temporary use.
For more details, refer to the following SciPy documentation:

> - https://docs.scipy.org/doc/scipy/dev/api-dev/array_api.html#using-array-api-standard-support

Author: > Note this is in principle a temporary thing in scipy while array API support is being worked on but I think this is worth mentioning anyway because this temporary thing may last a bit of time.

I found the same remark in SciPy's documentation. I hope adding the reference will be helpful.

> For more details, refer to the following SciPy documentation:
> 
> - https://docs.scipy.org/doc/scipy/dev/api-dev/array_api.html#using-array-api-standard-support

Reviewer 2: Can the link be incorporated into the text?

```suggestion
For more details, refer to SciPy's
`Array API documentation <https://docs.scipy.org/doc/scipy/dev/api-dev/array_api.html#using-array-api-standard-support>`__.
```

    Currently, only `array-api-strict`, `cupy`, and `PyTorch` are known to work
    with scikit-learn's estimators.

The following video provides an overview of the standard's design principles
and how it facilitates interoperability between array libraries:

- `Scikit-learn on GPUs with Array API <https://www.youtube.com/watch?v=c_s8tr1AizA>`_

>   by `Thomas J. Fan`_ at PyData NYC 2023.

Reviewer: Fine with having the video, although at the same time it can be slightly outdated since array API is evolving.

I guess use `:user:` as in the changelog:
```suggestion
  by :user:`Thomas Fan <thomasjpfan>` at PyData NYC 2023.
```

> 

Author: I hope this YouTube video can help end-users when using the Array API. If not, I can have it removed.

scikit-learn will revert to using the `float32` data type instead. This can result in
different behavior (typically numerically unstable results) compared to not using array
API dispatching or using a device with `float64` support.

> .. _Thomas J. Fan: https://github.com/thomasjpfan

Reviewer: ```suggestion
```
